lib/command/update/general_lastup_updater.rb

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2013 whiteleaf. All rights reserved.
+#
+
+require_relative "../../narou/api"
+
+module Command
+  class Update < CommandBase
+    class GeneralLastupUpdater
+      def initialize(options)
+        @options = options
+        @database = Database.instance
+        @narou_novels = Hash.new { |h, k| h[k] = [] }
+        @other_novels = []
+        prepare
+      end
+
+      def save
+        @database.save_database
+      end
+
+      def prepare
+        @database.each.with_index do |(id, data), i|
+          next if Narou.novel_frozen?(id)
+          setting =  Downloader.get_sitesetting_by_target(id)
+          if setting["narou_api_url"]
+            @narou_novels[setting["narou_api_url"]] << setting["ncode"]
```

Comment:
__[reek]__
* [FeatureEnvy] refers to setting more than self (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md)

<!-- Comment by [SideCI](https://sideci.com) -->